packages/firestore/test/unit/specs/spec_test_components.ts

```diff
@@ -398,7 +404,10 @@ export class MockConnection implements Connection {
           ++this.watchStreamRequestCount;
           if (request.addTarget) {
             const targetId = request.addTarget.targetId!;
-            this.activeTargets[targetId] = request.addTarget;
+            this.activeTargets[targetId] = {
```

Comment:
Record the actual listen request label to validate targetPurpose in active targets.